src/server/game/Handlers/ItemHandler.cpp

```diff
@@ -440,6 +440,13 @@ void WorldSession::HandleSellItemOpcode(WorldPacket& recvData)
         ItemTemplate const* pProto = pItem->GetTemplate();
         if (pProto)
         {
+            uint32 money = pProto->SellPrice * count;
+            if (_player->GetMoney() + money > MAX_MONEY_AMOUNT)               // prevent exceeding gold limit
+            {
+                _player->SendEquipError(EQUIP_ERR_TOO_MUCH_GOLD, nullptr, nullptr);
+                _player->SendSellError(SELL_ERR_UNK, creature, itemguid, 0);
+                return;
+            }
```

Comment:
this code should be in the `if (pProto->SellPrice > 0)` body.

Reply:
ok, i will move it